modules/setting/setting.go

```diff
@@ -1045,12 +1044,12 @@ func loadOrGenerateInternalToken(sec *ini.Section) string {
 	return token
 }
 
-func makeManifestData() template.URL {
-	name := url.QueryEscape(AppName)
-	prefix := url.QueryEscape(StaticURLPrefix)
-	subURL := url.QueryEscape(AppSubURL) + "/"
+func makeManifestData() string {
+	name := util.JSONEscape(AppName)
+	prefix := util.JSONEscape(StaticURLPrefix)
+	subURL := util.JSONEscape(AppSubURL) + "/"
 
-	return template.URL(`data:application/json,{"short_name":"` + name + `","name":"` + name + `","icons":[{"src":"` + prefix + `/img/logo-lg.png","type":"image/png","sizes":"880x880"},{"src":"` + prefix + `/img/logo-sm.png","type":"image/png","sizes":"120x120"},{"src":"` + prefix + `/img/logo-512.png","type":"image/png","sizes":"512x512"},{"src":"` + prefix + `/img/logo-192.png","type":"image/png","sizes":"192x192"}],"start_url":"` + subURL + `","scope":"` + subURL + `","background_color":"%23FAFAFA","display":"standalone"}`)
+	return `application/json;base64,` + base64.StdEncoding.EncodeToString([]byte(`{"short_name":"`+name+`","name":"`+name+`","icons":[{"src":"`+prefix+`/img/logo-lg.png","type":"image/png","sizes":"880x880"},{"src":"`+prefix+`/img/logo-sm.png","type":"image/png","sizes":"120x120"},{"src":"`+prefix+`/img/logo-512.png","type":"image/png","sizes":"512x512"},{"src":"`+prefix+`/img/logo-192.png","type":"image/png","sizes":"192x192"}],"start_url":"`+subURL+`","scope":"`+subURL+`","background_color":"%23FAFAFA","display":"standalone"}`))
```

Comment:
Does the order of these values matter?

* if not it might be nicer to use a `map[string]interface{}` here and marshal it.
* if so, it might similar be nicer to use a temporary struct: 

```go

func makeManifestData() string {
	name := AppName
	prefix := StaticURLPrefix
	subURL := AppSubURL + "/"
  
  type manifestJson struct {
    ShortName string `json:"short_name"`
    Name string `json:"name"`
    ...
  }

  bs, err := json.Marshal(&manifestJson{
    ...
  })

  if err != nil {
    log.Error("unable to marshal manifest JSON. Error: %v", err)
  }

  return `application/json;base64,`+base64.StdEncoding.EncodeToString(bs)
```

Reply:
Order should be insignificant but I'd still prefer something that retains it, so I guess I'll go with your second solution for readability.

Reply:
Did this